runtime/compiler/trj9/env/J9CPU.hpp

```diff
@@ -45,6 +46,14 @@ class CPU : public OMR::CPUConnector
 public:
 
    J9ProcessorDesc *TO_PORTLIB_getJ9ProcessorDesc();
+
+   // The following methods should not be called in the base class here
+   // They are just here so that these x86 specific methods have a declaration/definition during compilation
+   // x86 specific functions:
+   bool getX86SupportsSSE4_1() { TR_ASSERT(false, "Should not be called!"); return false; }
+   const char *getX86ProcessorVendorId() { TR_ASSERT(false, "Should not be called!"); return NULL; }
+   uint32_t getX86ProcessorSignature() { TR_ASSERT(false, "Should not be called!"); return 0; }
```

Comment:
is there a reason these are going in J9CPU and not pushed all the way down to OMRCPU in the OMR project?

Reply:
These 3 functions contain call sites that are exposed to all platforms, which means for non-X platforms, these functions won't have a declaration/definition due to its x/J9CPU class not being there. So this is just a workaround.